packages/jellyfish-api-core/__tests__/category/loan/listAuctons.test.ts

```diff
@@ -0,0 +1,481 @@
+      expect(pageLimit4[1].vaultId).toStrictEqual(auction2.vaultId)
+      expect(pageLimit4[2].vaultId).toStrictEqual(auction3.vaultId)
+      expect(pageLimit4[3].vaultId).toStrictEqual(auction4.vaultId)
+
+      // List auctions with limit > size
+      const pageLimit5 = await testing.rpc.loan.listAuctions({ limit: 5 })
+      expect(pageLimit5.length).toStrictEqual(4)
+      expect(pageLimit5).toStrictEqual(pageLimit4)
+    })
+
+    it('should listAuctions with height', async () => {
+      // List for liquidation height of first vault
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction1.liquidationHeight }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(4)
+        expect(page[0].vaultId).toStrictEqual(auction1.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction2.vaultId)
+        expect(page[2].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[3].vaultId).toStrictEqual(auction4.vaultId)
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction1.liquidationHeight } } // Default = including_start = false
+        )
+        expect(page.length).toStrictEqual(3)
+        expect(page[0].vaultId).toStrictEqual(auction2.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[2].vaultId).toStrictEqual(auction4.vaultId)
+      }
+
+      // List for liquidation height of forth vault
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction4.liquidationHeight }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(1)
+        expect(page[0].vaultId).toStrictEqual(auction4.vaultId)
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction4.liquidationHeight } }
+        )
+        expect(page.length).toStrictEqual(0)
+      }
+    })
+
+    it('should listAuctions with vaultId', async () => {
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction1.vaultId }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(4) // Unable to filter by vaultId only. Need to filter both vaultId and height together.
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction1.vaultId } }
+        )
+        expect(page.length).toStrictEqual(3) // Unable to filter by vaultId only. Need to filter both vaultId and height together.
+      }
+    })
+
+    it('should listAuctions with vaultId and height', async () => {
+      // List for liquidation height of second vault and vaultId of second vault
```

Comment:
I do not add the test for liquidation height and vaultId that belongs to different vault (For eg: liquidation height of third vault and vaultId of second vault).  The reason is the result is unpredictable where different result is returned if you run the test for many times.

Reply:
not sure the reason for this. But when using height and vault id from different auctions, basically the entry point that we are giving is non-existent. Not sure how it behaves with a combined index.